lib/at_cmd/at_cmd.c

```diff
@@ -22,34 +22,44 @@ LOG_MODULE_REGISTER(at_cmd, CONFIG_AT_CMD_LOG_LEVEL);
 #define AT_CMD_CMS_STR   "+CMS ERROR:"
 #define AT_CMD_CME_STR   "+CME ERROR:"
 
-static K_THREAD_STACK_DEFINE(socket_thread_stack, \
-				CONFIG_AT_CMD_THREAD_STACK_SIZE);
+/* Flags describing an AT command request */
+enum at_cmd_flags {
+	AT_CMD_BUF_CMD = 1 << 0,	/* Command is buffered by at_cmd */
+	AT_CMD_SYNC = 1 << 1,		/* Command is synchronous */
+};
+
+/* Metadata for a queued AT command */
+struct cmd_item  {
+	char *cmd;			/* Pointer to 0-terminated command */
+	char *buf;			/* Pointer to response buffer */
```

Comment:
You could maybe rename `buf` and `buf_len` to `resp` and `resp_len`, and group them in the struct?
I assume the response is also a null-terminated String?

Reply:
Done. The grouping is to save memory